lightning/src/ln/msgs.rs

```diff
@@ -827,6 +827,25 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn get_next_node_announcements(&self, starting_point: Option<&PublicKey>, batch_amount: u8) -> Vec<NodeAnnouncement>;
 	/// Returns whether a full sync should be requested from a peer.
 	fn should_request_full_sync(&self, node_id: &PublicKey) -> bool;
+	/// Initiates routing gossip sync by querying a peer to discover channels
+	/// and their associated routing gossip messages. This method will use a
+	/// sync strategy defined by the implementor.
+	fn sync_routing_table(&self, their_node_id: &PublicKey);
```

Comment:
It may be useful to pass in the `Init` message here so that the NetGraphMessageHandler can analyze the InitFeatures. Same goes for `should_request_full_sync` - that way we can only return true in the second when the peer doesn't support gossip queries and send a query_channel_range when they do.